client/extensions/woocommerce/index.js

```diff
@@ -265,9 +266,19 @@ export default async function ( _, addReducer ) {
 	// Add pages that use the store navigation
 	getStorePages().forEach( function ( storePage ) {
 		if ( config.isEnabled( storePage.configKey ) ) {
-			addStorePage( storePage, ( context, next ) =>
-				createStoreNavigation( context, next, storePage )
-			);
+			// Store deprecation would redirect most store pages to dashboard
+			if ( isStoreRemoved && ! ( '/store/:site' === storePage.path ) ) {
+				page( storePage.path, addTracksContext, ( context ) => {
+					// Tracks MC stats by path /store/products/:site -> store-products-:site
+					const trackName = storePage.path.substr( 1 ).replaceAll( '/', '-' );
+					context.store.dispatch( bumpStat( 'calypso_store_post_sunset', trackName ) );
+					page.redirect( `/store/${ context.params.site }?${ context.querystring }` );
```

Comment:
Redirect with GET query parameters to preserve any context that a user might have saved in their bookmark etc.